exercise_1.js

```diff
@@ -0,0 +1,88 @@
+// Import modules
+
+const express = require('express');
+const app = express();
+
+const bodyParser = require('body-parser');
+const replaceString = require('replace-string');
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended : false}));
+
+//post method
+app.post('/',(req,res)=>
+{
+
+	let input = req.body;
+	let myJSON = JSON.stringify(input);
+	let ref=(input.referenceData);
+	let jsonPayload = JSON.stringify(input.payload);
+
+//All reference objects are push into an array
+	let listOfObj=[];
+	Object.keys(req.body.referenceData).forEach((key,index)=>
+	{
+	let modi="{"+ key +"}";
+	let value=req.body.referenceData[key];
+
+		listOfObj.push({[modi]:value});
+
+	});
+
+//Array objects are converted into single object
+let obj = Object.assign({},...listOfObj);
+
+//objects values are mapped and replaced
+let RE = new RegExp(Object.keys(obj).join("|"), "gi");
+		myJSON = jsonPayload.replace(RE,(matched)=>
+		{
+return obj[matched];
+
+		});
+
+input=JSON.parse(myJSON);
+
+if(!input || input.text ==="")
+{
+    let mapDictionary = new Map();
+
+	for(x in ref)
+	{
+		mapDictionary.set(x,ref[x]);
+	}
+
+	let itr=mapDictionary.keys();
+
+	for(let i=0;i<mapDictionary.size;i++)
+	{
+		const key=itr.next().value;
+
+		myJSON = replaceString(myJSON, '{'+ key +'}', mapDictionary.get(key));
```

Comment:
can you test with bigger payload? say 4MB, 10MB etc? and see if the code works without the any system hang?